### ISSUE TITLE
`Label` text can cause page to crash

### ISSUE THREAD

**Issue by Cameramorphic** [amcharts/amcharts5#992]
**Bug description**
The text content of a `Label` can cause the whole browser page to crash. It seems like amCharts is getting stuck in an endless loop.
This happens when the label text is e.g. `" "` or `"???"`, we could also narrow it down that it happens when certain characters are at the start of any text e.g. `"?test text"`.
We are using axis range labels as described in the [Axis ranges documentation](https://www.amcharts.com/docs/v5/charts/xy-chart/axes/axis-ranges/), with the following settings:
```ts
rangeDataItem.get('label')?.setAll({
    text: labelText,
    maxWidth: 300,
    oversizedBehavior: 'wrap',
    textAlign: 'center',
```
It is more complex than this, including an onChange handler for setting `maxWidth` dynamically, but this is a minimal example.
In our application the label text is user-configurable, so this is particularly problematic, because users could run into this issue at any time.

Maybe this is related to issue #351, as it looks like the label is the culprit there too?

**Environment**
amCharts: 5.4.0
Browser: Chrome 114.0.5735.198
Frameworks: Typescript 4.9.5, React 18.2.0

**Comment by martynasma** [comment]
Any hints on how we can reproduce this?

I tried adding labels like that to axis range. It does not crash:
https://codepen.io/team/amcharts/pen/poQrYKR/6f22c26f7facf78e6f9460864b0eb6e0?editors=0010

**Comment by Cameramorphic** [comment]
I just debugged again, you can reproduce it e.g. for `"???"` by setting `maxWidth` to `30` or lower.

I also checked, why this *always* happens in our code for seemingly bigger values of `maxWidth`, the reason has to do with the onChange handler that sets it:
```ts
chart.plotContainer.onPrivate('width', () => label?.set('maxWidth', rangeDataItem.get('axisFill')?.width());
```
It looks like `rangeDataItem.get('axisFill')?.width()` returns `0` on the first (few) render cycles in React, where it then crashes and doesn't continue to render.
(We are using the ranges on a category x-axis, but that isn't relevant to reproduce it).

I also managed to reproduce it in your example by using the label from #351 like this:
```ts
text: "[#000000]{value}[/]   [#000000]{value}% ({value}/200)[/]",
populateText: true,
maxWidth: 50,
```

**Comment by martynasma** [comment]
Thanks. I was able to reproduce it now. Will be looking into it.

**Comment by martynasma** [comment]
Fixed in [5.4.1](https://github.com/amcharts/amcharts5/releases/tag/5.4.1).

## [5.4.1] - 2023-07-18

### Added
- All Flow (Sankey, Chord and Arc) nodes can be toggled (on by default). [More info](https://www.amcharts.com/docs/v5/charts/flow-charts/#Node_toggling).
- New `Flow` settings added: `hiddenSize` and `minHiddenValue`. [More info](https://www.amcharts.com/docs/v5/charts/flow-charts/#Node_toggling).
- Deviation added to editable `BolingerBand` settings (`StockChart`).

### Changed
- `Chord`'s default value of `"sort"` changed from `"descending"` to `"none"`.  This change was necessary for the new feature of [toggling nodes](https://www.amcharts.com/docs/v5/charts/flow-charts/#Node_toggling).

### Fixed
- `Label` with wrapping/truncation, containing only symbols like question mark could make the whole chart break in a dead loop.
- `exclude`/`include` settings on `MapSeries` was being ignored when new data was set for the series.
- `LineSeries` was not being included in the caluclation of min/max of the vertical `ValueAxis` if it did not have any data points in current zoom scope of the horizontal `DateAxis`.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by Cameramorphic** [comment]
The issue I had is fixed, thanks!
However, if one wants to use `oversizedBehavior: 'wrap-no-break'` it is still possible to cause this infinite loop.
(might also happen in other cases where `fallbackBreakWords` ends up being `false` in `CanvasRenderer._truncateText`)
